ios/Codeforces Watcher/Application/features/problems/ProblemsTableViewAdapter.swift

```diff
@@ -13,43 +13,63 @@ import common
 class ProblemsTableViewAdapter: NSObject, UITableViewDelegate, UITableViewDataSource {
     
     var problems: [Problem] = []
+    var favouriteProblems: [Problem] = []
```

Comment:
I see multiple code-smells here:
- `TableViewAdapter` knows about `store`
- `problems` and `favouriteProblems` server exactly the same purpose, but treated the same way
- `isFavourite` is used in at least 5 places

Generally speaking, you've put the logic, which should exist outside `TableViewAdapter`, inside it. Check the Android implementation and how elegantly it's organized in `ProblemsFragment.newState()`.